### DISCUSSION TITLE
Unable to Execute Go Serverless Function

### DISCUSSION THREAD

**Discussion by jcald1** [vercel/next.js#15785] [Help] [answered]
I've created an app with a JS (Node.js) serverless function in /pages/api/seed-data.js. I created a PR and on the Vercel CI server (https://commercial-real-estate-datadeck-jc-git-feature-go-basic.jcald1.vercel.app/), I'm able to access that API route. (<host>/api/seed-data)

I also included the Go sample file from https://vercel.com/docs/v2/serverless-functions/supported-languages#go in that same directory.

I expected that a call to <host>/api/date to work.  However, I'm getting a 404.

As a related question, is it possible to test such a serverless endpoint locally?  Do I need to run `vercel dev` and have installed Go locally on the host first?  Or would it run the Go code in a container so the host doesn't need Go to be installed?

**Comment by leo** [comment, marked as answer]
Thank you for sharing this issue with us!

Next.js natively only supports Node.js Serverless Functions. Those are the ones you place in `pages/api`.

If you want to write Serverless Functions in Node.js _and_ Go, Next.js won't help there, unfortunately. You'd have to place the Serverless Functions inside the `api` (not `pages/api`) directory, which makes Vercel take care of them.

And yes, then you'd have to run `vercel dev`. You can read more about how to use Serverless Functions on Vercel [here](https://vercel.com/docs/v2/serverless-functions/introduction).
